code/online_challenges/src/project_euler/problem_037/problem_037.cpp

```diff
@@ -0,0 +1,59 @@
+#include <array>
+#include <cmath>
+#include <iostream>
+
+template<std::size_t N>
+std::array<bool, N> primesUpto() // Function that implements the Sieve of Eratosthenes
+{
+    std::array<bool, N> primesList;
+    
+    std::fill(primesList.begin(), primesList.end(), true);
+    
+    primesList[0] = primesList[1] = false;
+    
+    std::size_t sqrtLimit = std::sqrt(N) + 1;
+    
+    for (std::size_t i = 0; i < sqrtLimit; ++i)
+        if (primesList[i])
+            for (std::size_t j = i + i; j < N; j += i)
+                primesList[j] = false;
+    
+    return primesList;
+}
+
+template<std::size_t N>
+bool isTruncPrime(std::size_t number, const std::array<bool, N>& primesList)
+{
+    for (std::size_t i = 10; i < number; i *= 10)
+        if (primesList[number % i] == 0) // If the right truncated part is not prime
+            return false;
+    
+    for (number; number >= 1; number /= 10)
+        if (primesList[number] == 0) // If the left truncated part is not prime
+            return false;
+    
+    return true; // All truncated parts are prime, so the number is a truncatable prime
+}
+
+int main()
+{
+    const auto primesUptoMillion = primesUpto<1000000ULL>(); // Represents all the primes up to 1 million
+    
+    std::size_t numberTruncatablePrimes = 0;
+    std::size_t currentNumber = 11; // 2, 3, 5, and 7 are not included in the search for truncatable primes
+    std::size_t truncatablePrimeSum = 0;
+    
+    while (numberTruncatablePrimes != 11) 
+    {
+        if (
+            primesUptoMillion[currentNumber] && // If the number itself is prime
```

Comment:
The logical operator is placed at end that seems more beautiful than begin, because every line will be aligned.
What do you think of?

Reply:
@iattempt I don't get what you are saying. Are you suggesting we change the style guide?

Reply:
In our current coding style, the `&&` is should be write at front of second condition.

Reply:
But I think your coding-style is more beautiful than it, despite readability.
Thus, I want to change it.

Reply:
Ok, I'll do it @iattempt.